src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -718,6 +712,13 @@ private void ExecuteOperationAction(
                 cancellationToken);
         }
 
+        private readonly record struct ExecutionData(
```

Comment:
The compiler's use of records is very restricted. It doesn't seem like we're making use of `Equals` on this new type, so it should just be a regular struct.

Reply:
sure.  i can do that.  though it seems very verbose.  :)

Reply:
done.

Reply:
Sorry, it needs to be even more verbose ;-) We also don't do primary constructor captures. 
This should have been flagged by an analyzer: [https://github.com/dotnet/roslyn-analyzers/pull/7138](https://github.com/dotnet/roslyn-analyzers/pull/7138) 
Tagging @333fred to confirm

Reply:
what is the value in that verbosity?  like... at least with records i can see not wanting the IL/metadata.  But really this seems to be adding noise and hiding clear concepts in lots of code.

Reply:
note: the analyzer you are referencing disallows implicit capture.  there is no implicit capture.  it's all explicitly written to the explicit fields.